File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListContract.java
package fr.free.nrw.commons.contributions;

import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
import fr.free.nrw.commons.BasePresenter;

/**
 * The contract for Contributions list View & Presenter
 */
public class ContributionsListContract {

    public interface View {

        void showWelcomeTip(boolean numberOfUploads);

        void showProgress(boolean shouldShow);

        void showNoContributionsUI(boolean shouldShow);
    }

    public interface UserActionListener extends BasePresenter<View> {

        void refreshList(SwipeRefreshLayout swipeRefreshLayout);
    }
}

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListContract.kt
package fr.free.nrw.commons.contributions

import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import fr.free.nrw.commons.BasePresenter

/**
 * The contract for Contributions list View & Presenter
 */
class ContributionsListContract {
    interface View {
        fun showWelcomeTip(numberOfUploads: Boolean)

        fun showProgress(shouldShow: Boolean)

        fun showNoContributionsUI(shouldShow: Boolean)
    }

    interface UserActionListener : BasePresenter<View> {
        fun refreshList(swipeRefreshLayout: SwipeRefreshLayout?)
    }
}
